src/iter/unzip.rs

```diff
@@ -386,3 +395,57 @@ impl<A, B, RA, RB> Reducer<(A, B)> for UnzipReducer<RA, RB>
         (self.left.reduce(left.0, right.0), self.right.reduce(left.1, right.1))
     }
 }
+
+
+impl<A, B, FromA, FromB> ParallelExtend<(A, B)> for (FromA, FromB)
```

Comment:
FWIW, the standard library does not have an `Extend` implementation like this, though I think it could.  The value there would not be as high though, as there's significantly more magic involved in the parallel implementation.